pkg/asset/machines/master.go

```diff
@@ -606,3 +610,42 @@ func IsMachineManifest(file *asset.File) bool {
 		return matched
 	}
 }
+
+func generateSecretAssets(resources []corev1.Secret, fileName string) (assets []*asset.File, err error) {
+
+	var objects []interface{}
+	for _, r := range resources {
+		objects = append(objects, r)
+	}
+
+	return generateAssets(objects, fileName)
+}
+
+func generateHostAssets(resources []baremetalhost.BareMetalHost, fileName string) (assets []*asset.File, err error) {
+
+	var objects []interface{}
+	for _, r := range resources {
+		objects = append(objects, r)
+	}
+
+	return generateAssets(objects, fileName)
+}
+
+func generateAssets(objects []interface{}, fileName string) (assets []*asset.File, err error) {
```

Comment:
The use of named result parameters here does not add anything to document the behavior of the function. As such, please refrain from naming the result parameters.